core/trino-main/src/test/java/io/trino/operator/aggregation/state/BenchmarkLongDecimalWithOverflowAndLongStateSerializer.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator.aggregation.state;
+
+import com.google.common.collect.ImmutableSet;
+import io.trino.jmh.Benchmarks;
+import io.trino.operator.aggregation.state.LongDecimalWithOverflowAndLongStateFactory.GroupedLongDecimalWithOverflowAndLongState;
+import io.trino.spi.block.Block;
+import io.trino.spi.block.BlockBuilder;
+import io.trino.spi.function.AccumulatorStateSerializer;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+import org.testng.annotations.Test;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.IntStream;
+
+import static com.google.common.base.Verify.verify;
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static java.util.stream.Collectors.toList;
+
+@State(Scope.Thread)
+@OutputTimeUnit(TimeUnit.MICROSECONDS)
+@Fork(2)
+@Warmup(iterations = 10)
+@Measurement(iterations = 20)
+@BenchmarkMode(Mode.AverageTime)
+public class BenchmarkLongDecimalWithOverflowAndLongStateSerializer
+{
+    @Benchmark
+    public Object serialize(BenchmarkData data)
+    {
+        return data.serialize(new LongDecimalWithOverflowAndLongStateSerializer());
+    }
+
+    @Benchmark
+    public Object deserialize(BenchmarkData data)
+    {
+        return deserialize(data, new LongDecimalWithOverflowAndLongStateSerializer());
+    }
+
+    private GroupedLongDecimalWithOverflowAndLongState deserialize(BenchmarkData data, AccumulatorStateSerializer<LongDecimalWithOverflowAndLongState> serializer)
```

Comment:
either have both `serialize` and `derialize` as separate method or method within `BenchmarkData`

Reply:
moved `deserialize` to `BenchmarkData`